src/topp/PercolatorAdapter.cpp

```diff
@@ -452,16 +470,24 @@ class PercolatorAdapter :
         
         int charge = hit.getCharge();
         String unmodified_sequence = hit.getSequence().toUnmodifiedString();
-        
-        double calc_mass = hit.getSequence().getMonoWeight(Residue::Full, charge)/charge;
-        hit.setMetaValue("CalcMass", calc_mass);
+       
+        double calc_mass; 
+        if (!hit.metaValueExists("CalcMass"))
+        {
+          calc_mass = hit.getSequence().getMonoWeight(Residue::Full, charge)/charge;
+          hit.setMetaValue("CalcMass", calc_mass);
+        }
+        else
+        {
+          calc_mass = hit.getMetaValue("CalcMass");
+        }
 
         if (hit.metaValueExists("IsotopeError"))  // MSGFPlus
         {
           float isoErr = hit.getMetaValue("IsotopeError").toString().toFloat();
           exp_mass = exp_mass - (isoErr * Constants::C13C12_MASSDIFF_U) / charge;
         }
-        else if (hit.metaValueExists("isotope_error")) // e.g. SimpleSearchEngine /RNPxlSearch
+        else if (hit.metaValueExists("isotope_error")) // e.g. SimpleSearchEngine /OpenNuXL
```

Comment:
Can't we unify this MetaValue?

Reply:
you mean put it here https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/CONCEPT/Constants.cpp#L173

Reply:
the thing is that MSGFPlus uses IsotopeError and we usually the lower case names.

Reply:
And why?

Reply:
maybe add isotope_error to the Constants file and make MSGFPlus adapter name it IsotopeError

Reply:
https://github.com/OpenMS/OpenMS/blob/d17cc251fd0c4068eb253b03c9fb107897771fdc/src/openms/source/ANALYSIS/ID/PercolatorFeatureSetHelper.cpp#L49

https://github.com/OpenMS/OpenMS/blob/c6348e527cd58d7bad637fe8849deae4ca7a2ed2/src/topp/MSGFPlusAdapter.cpp#L608

https://github.com/OpenMS/OpenMS/blob/c6348e527cd58d7bad637fe8849deae4ca7a2ed2/src/topp/PercolatorAdapter.cpp#L459

These seem to be the only occurances

Reply:
Yes, I would say stick to one for writing: But for now, allow both for reading.